Win/SketchfabUploader.h

```diff
@@ -0,0 +1,233 @@
+
+        std::string value((success ? model_url(get_json_key(response.second, "uid")) : get_json_key(response.second, "detail")));
+
+        return std::pair<bool, std::string>(success, value);
+    }
+
+
+    std::pair<int, std::string> post(const std::string& url,
+                                     const attributes& files = attributes(),
+                                     const attributes& parameters = attributes(),
+                                     HWND progressBar=NULL)
+    {
+        CURL *curl;
+        struct ProgressbarUpdater prog;
+        CURLcode res;
+        long http_code;
+        std::string response;
+        struct curl_httppost *formpost = NULL;
+        struct curl_httppost *lastptr = NULL;
+        http_code = 0;
+
+        if (url.empty()) {
+            std::cout << "Missing url parameter in post parameters" << std::endl;
+            return std::pair<int, std::string>(-1, std::string());
+        }
+        if (files.empty()) {
+            std::cout << "Missing files in post parameters" << std::endl;
+            return std::pair<int, std::string>(-2, std::string());
+        }
+
+        for (attributes::const_iterator file = files.begin(); file != files.end(); ++file) {
+            std::cout << file->first.c_str();
+			curl_formadd(&formpost,
+				         &lastptr,
+				         CURLFORM_COPYNAME, file->first.c_str(),
+				         CURLFORM_FILE, file->second.c_str(),
+				         CURLFORM_END);
+        }
+
+        for (attributes::const_iterator parameter = parameters.begin(); parameter != parameters.end(); ++parameter) {
+              if (!parameter->second.empty()) {
+                curl_formadd(&formpost,
+                             &lastptr,
+                             CURLFORM_COPYNAME, parameter->first.c_str(),
+                             CURLFORM_COPYCONTENTS, parameter->second.c_str(),
+                             CURLFORM_END);
+              }
+        }
+
+    // Setting curl
+    curl = curl_easy_init();
+
+    if (curl) {
+        prog.curl = curl;
+        prog.progressBar = progressBar;
+
+        curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteMemoryCallback);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response);
+        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+        curl_easy_setopt(curl, CURLOPT_HTTPPOST, formpost);
+        curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0);
+        curl_easy_setopt(curl, CURLOPT_PROGRESSDATA, &prog);
+        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 900L);
+        curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION, progress_callback);
+        res = curl_easy_perform(curl);
+        if (res != CURLE_OK) {
+            return std::pair<int, std::string>(1, "{\"detail\":\"curl_easy_perform() failed: " + std::string(curl_easy_strerror(res)) + "\" } ");
+            std::cerr << "curl_easy_perform() failed: " << curl_easy_strerror(res) << std::endl;
```

Comment:
Please comment out or delete these two lines, since the "return" means they'll never be executed.